actgen/agents/dqn.py

```diff
@@ -8,18 +8,19 @@
 
 
 class DQNAgent():
-    def __init__(self, observation_space, action_space, params):
-        self.observation_space = observation_space
-        self.action_space = action_space
+    def __init__(self, env, params):
+        self.env = env
```

Comment:
I usually try to keep the env and the agent fully separated. Is there a reason the agent needs to have access to the full env?

If this change was to compensate for the fact that now the env knows which actions are duplicates, maybe we can simply pass the `get_duplicate_actions()` function as a callable argument alongside the `observation_space` and `action_space`?

Reply:
yea that's exactly the reason. Good point! I'll keep the env and agent fully separated.